File: app/src/main/java/com/gplio/fibrewallpaper/TestActivity.java
package com.gplio.fibrewallpaper;

import android.Manifest;
import android.content.pm.PackageManager;
import android.opengl.GLSurfaceView;
import android.os.Environment;
import android.support.v4.content.ContextCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;

import com.gplio.andlib.files.ExternalFileObserver;
import com.gplio.andlib.graphics.DebugText;
import com.gplio.fibrewallpaper.main.MainRenderer;

public class TestActivity extends AppCompatActivity {

    private MainRenderer renderer;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test);

        GLSurfaceView surface = findViewById(R.id.surface_view);

        renderer = new MainRenderer(this);
        surface.setPreserveEGLContextOnPause(true);
        surface.setEGLContextClientVersion(2);
        surface.setEGLConfigChooser(8, 8, 8, 8, 16, 0);
        surface.setRenderer(renderer);
        surface.setRenderMode(GLSurfaceView.RENDERMODE_CONTINUOUSLY);



        if (ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)
                != PackageManager.PERMISSION_GRANTED) {
            throw new RuntimeException("Please grant read/write storage permission");
        }
    }

    @Override
    protected void onResume() {
        super.onResume();
        log("onResume");

        DebugText.parseEasyFont(this);
    }

    @Override
    protected void onPause() {
        super.onPause();
        log("onPause");
        renderer.unsubscribeExternalEvents();
    }

    private static void log(String msg) {
        Log.d("TestActivity", msg);
    }
}

File: app/src/main/java/com/gplio/fibrewallpaper/TestActivity.kt
package com.gplio.fibrewallpaper

import android.content.IntentFilter
import android.opengl.GLSurfaceView
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.gplio.fibrewallpaper.lib.logger.Tagged
import com.gplio.fibrewallpaper.lib.logger.d
import com.gplio.fibrewallpaper.lib.observers.ShaderChangeObserver
import com.gplio.fibrewallpaper.main.ChangesBroadcastReceiver
import com.gplio.fibrewallpaper.main.MainRenderer
import com.gplio.fibrewallpaper.values.ACTION_UPDATE_SHADERS

class TestActivity : AppCompatActivity(), ShaderChangeObserver, Tagged {

    override val tag: String
        get() = "TestActivity"

    private val changesBroadcastReceiver = ChangesBroadcastReceiver()

    private var renderer: MainRenderer? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_test)

        val surface = findViewById<GLSurfaceView>(R.id.surface_view)

        renderer = MainRenderer(this@TestActivity)

        surface.preserveEGLContextOnPause = true
        surface.setEGLContextClientVersion(2)
        surface.setEGLConfigChooser(8, 8, 8, 8, 16, 0)
        surface.setRenderer(renderer)
        surface.renderMode = GLSurfaceView.RENDERMODE_CONTINUOUSLY

        d("onCreate", "registering receiver")

        IntentFilter(ACTION_UPDATE_SHADERS).also {
            registerReceiver(changesBroadcastReceiver, it)
        }
    }

    override fun onResume() {
        super.onResume()
        d("onResume")

        changesBroadcastReceiver.setListener(this)
    }

    override fun onPause() {
        super.onPause()
        d("onPause")

        // changesBroadcastReceiver.setListener(null)
    }

    override fun onDestroy() {
        super.onDestroy()
        d("onDestroy")

        //unregisterReceiver(changesBroadcastReceiver)
    }

    override fun onShadersChanged(vertexShader: String, fragmentShader: String) {
        renderer?.requestShaderUpdate(vertexShader, fragmentShader)
    }
}